docs/set-up.asciidoc

```diff
@@ -35,36 +36,284 @@ include::./restify.asciidoc[]
-This means that you should probably require and start it in your applications main file (usually `index.js`, `server.js`, or `app.js`).
+There are a few ways to start the Node.js APM agent. Choose the one that works best for you. The most important considerations for selecting a method are:
+
+- ensuring the APM agent starts early enough, and
+- having a convenient way to configure the agent.
+
+For the Node.js APM agent to be able to fully function, it *must be started before `require(...)` statements for other modules*. The APM agent automatically instruments modules by interposing itself in the import process. If a given module is imported before the APM agent has started, then it won't be able to instrument that module.
+
+
+==== Start methods
+
+[[start-option-require-and-start]]
+===== `require('elastic-apm-node').start(...)`
+
+The most common way to start the APM agent is to require the `elastic-apm-node` module and call the <<apm-start,`start`>> function at the top of your main module. This allows you to use any of the methods to <<configuring-the-agent,configure the agent>>.
+
+[source,js]
+----
+const apm = require('elastic-apm-node').start({
+  // Add configuration options here.
+});
+
+// Application main code goes here.
+----
+
+
+[[start-option-require-start-module]]
+===== `require('elastic-apm-node/start')`
+
+Another way to start the agent is with the `elastic-apm-node/start` module that imports and _starts_ the agent.
+
+[source,js]
+----
+const apm = require('elastic-apm-node/start');
+
+// Application main code goes here.
+----
 
-There are two ways to start the Elastic APM agent for Node.js:
+This start method exists for those that use a tool like Babel or esbuild to translate/transpile from code using ES modules (as in the following example) to code using CommonJS. It ensures that the APM agent is started before other imports in the same file. See <<start-esm-imports>> below for details.
 
-* Require the `elastic-apm-node` module and call the <<apm-start,`start`>> function on the returned agent:
-+
 [source,js]
 ----
-var apm = require('elastic-apm-node').start({
-  // add configuration options here
+import 'elastic-apm-node/start';
+
+// Application main code goes here.
+----
+
+A limitation of this approach is that you cannot configure the agent with an options object, but instead have to rely on <<configuring-the-agent,one of the other methods of configuration>>, such as setting `ELASTIC_APM_...` environment variables.
+
+
+[[start-option-node-require-opt]]
+===== `node -r elastic-apm-node/start ...`
+
+Another way to start the agent is with the `-r elastic-apm-node/start` https://nodejs.org/api/cli.html#-r---require-module[command line option to `node`]. This will import and start the APM agent before your application code starts. This method allows you to enable the agent _without touching any code_. This is the recommended start method for <<lambda,monitoring AWS Lambda functions>>.
```

Comment:
This anchor doesn't work for me—I get linked to the bottom of the page. Weirdly, it does seem correct, and it does work when clicked in the Node.js docs.

Reply:
Hmm. It works in incognito. Must be something on my end.